### PR TITLE
Enabling updates to accept empty strings for User's profile for optional fields

### PR DESCRIPTION
**Pull request by kishansagathiya**
Made changes such that gorm can update fields with blank values as well
It was treating them as not-changed before

Fixes(#186)

### CODE REVIEW

**Comment by fabric8cd** [comment]
@kishansagathiya snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-189-1`

**Comment by fabric8cd** [comment]
@kishansagathiya snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-189-2`

**Comment by kishansagathiya** [comment]
[test]

**Comment by kishansagathiya** [comment]
[test]

File: account/identity.go [hunk 1]
```diff
@@ -202,7 +202,7 @@ func (m *GormIdentityRepository) Lookup(ctx context.Context, username, profileUR
 func (m *GormIdentityRepository) Save(ctx context.Context, model *Identity) error {
 	defer goa.MeasureSince([]string{"goa", "db", "identity", "save"}, time.Now())
 
-	obj, err := m.Load(ctx, model.ID)
+	_, err := m.Load(ctx, model.ID)
```
Comment (alexeykazakov) [inline account/identity.go]: Do we really need to load it with the new approach? What happen if you do m.db.Save(model) for unknown model? It should generate the same not-found error I guess?

Comment (kishansagathiya) [reply account/identity.go]: Why should we be concerned with the not-found error? to run update we need user token, and if we have the user token that means the account exist, hence found.

Comment (kishansagathiya) [reply account/identity.go]: Should be safe to remove that error check

**Comment by fabric8cd** [comment]
@kishansagathiya snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-189-3`

**Pull request merged**